### PR TITLE
Support/live 5375 Stake banner test

### PR DESCRIPTION
**Pull request by ggilchrist-ledger**
### 📝 Description

Playwright tests for Staking banners and flows
- Ethereum Staking flow from portfolio page and verifying navigation to provider (Lido)
- Ethereum staking flow from asset page
- Ethereum staking flow from market page/market coin detail page

Also includes a fix courtesy of @cjordan-ledger that allows non-BTC coins to be used in the Market Coin detail page (fix made in the `api.mock.ts` file that allows for other cryptos to be searchable by id)

### ❓ Context

- **Impacted projects**: Playwright Tests/Staking buttons and flows
- **Linked resource(s)**: https://github.com/LedgerHQ/ledger-live/pull/2554

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/87659842/218105874-7d731de5-e202-46d1-8201-1d452f4253fe.mov

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx [hunk 1]
```diff
@@ -45,7 +45,12 @@ export function CurrencyList({ currencies, onCurrencySelect }: Props) {
             {({ index, style }: any) => {
               const currency = currencies[index];
               return (
-                <RowContainer key={index} style={style} onClick={() => onCurrencySelect(currency)}>
+                <RowContainer
+                  key={index}
+                  style={style}
+                  onClick={() => onCurrencySelect(currency)}
+                  data-test-id={`currency-row-${currency.name.toLowerCase()}`}
```
Comment (cjordan-ledger) [inline apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx:52]: is it worth doing `currency?.name?.toLowerCase()`? if you're confident they're always set then ignore this.

File: apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/component/ProviderItem.tsx [hunk 2]
```diff
@@ -36,7 +36,13 @@ const ProviderItem = ({ id, name, provider, infoOnClick, stakeOnClick }: ItemPro
   );
 
   return (
-    <Container pl={3} flow={1} onClick={stakeLink} py={4}>
+    <Container
+      pl={3}
+      flow={1}
+      onClick={stakeLink}
+      py={4}
+      data-test-id={`stake-provider-container-${provider.name.toLowerCase()}`}
```
Comment (cjordan-ledger) [inline apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/component/ProviderItem.tsx:44]: same q, is it worth doing `provider?.name?.toLowerCase()`?

**Review by cjordan-ledger — APPROVED** [review]
Looks good, left some minor comments which probably aren't necessary.

File: apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx (hunk 1 above)
Comment (ggilchrist-ledger) [reply apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx:52]: Yeah probably worth doing. However would the component even load without `currency`?

Comment (cjordan-ledger) [reply apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx:52]: Possibly not. I'm happy for this to go in as is

**Review by sarneijim — APPROVED** [review]
Not for this pr, but It would be good begin to use conventional commit messages. 
https://www.conventionalcommits.org/en/v1.0.0/

File: apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx (hunk 1 above)
Comment (ggilchrist-ledger) [reply apps/ledger-live-desktop/src/renderer/drawers/DataSelector/CurrencyList.jsx:52]: I'll add it and test it. Having it in doesn't do any harm

**Comment by ggilchrist-ledger** [comment]
> 

Agreed, I started this PR before the new CI tools warn for bad commit messages. By conventional commit messages do you just mean prefixing them with `test:`, `fix:`, etc, or do I need to add the scope, body, footer stuff too? @sarneijim

**Pull request merged**